### ISSUE TITLE
feature request or simply by-design

### ISSUE THREAD

**Issue by ruxbo** [VorpalBlade/keyboard-backlightd#17]
i'm using:
t440s shelf/stock i7, 8gb, 500gb
fedora 39 beta
gnome 45

since lenovo made the fn+spacebar for kbb, i've been looking for something like this
i found the _other_ but nothing like this
i'm no expert, i wish it was packaged like you said-- in rpm 😁
but following your steps was easy

my apologies i skipped the template
everything is working!
i love it bigtime
i enabled the keys, trackpoint and touchpad
each time i touch the pad, it lits up! very nice
i'll test the other events later
like opening the lid or plugging the power adapter

i've been a thinkpad user for so long, it's a bad habit of hitting fn+spacebar when i'm in low light condition to get the kbb activated
if i do that sort of by accident, i need to stop, start the service again
is that right?
i don't think it's a bug

anyways, thank you for this
i'm learning
the next t450, t460, t470, t480, t490, and even t14 i'll be setting up 😁 i know what to use
oh, i need to test the x1 yoga as well 🤪 for the touch screen😉
_yeah, perks of being a windows admin_ 🤣

**Comment by VorpalBlade** [comment]
> if i do that sort of by accident, i need to stop, start the service again
> is that right?
> i don't think it's a bug

I'm finding it a bit hard to follow you, sorry. But are you saying that fn+space results in the service stopping working? That is unexpected, it should cycle through the light levels. At least on my T480 that cycles through off/half/full. Maybe you just cycled it to off by mistake? Try hitting fn+space again?

If that is not it, I'll get back to you with instructions for how to capture detailed logs some time during next week (busy this weekend).

**Comment by ruxbo** [comment]
my apologies

i just tested it

if i do a manual cycle of fn+spacebar and never turn it back on, it interrupts the service(?) and wont behave normally

if i hit the fn+spacebar again to light it up catches on and back to normal-- it turns it off automatically
keys, pad, and stick lights up the backlight again as i hit them


i guess you can add in the readme that if one manually turns it off using fn+spacebar the service is interrupted and the service needs to be stopped then restarted again or simply hit fn+spacebar to manually turn it on and service will be in normal operation

perhaps this is just unique to me
please disregard


thank you

**Comment by ruxbo** [comment]
my apologies

i just tested it

if i do a manual cycle of fn+spacebar and never turn it back on, it interrupts the service(?) and wont behave normally

if i hit the fn+spacebar again to light it up catches on and back to normal-- it turns it off automatically
keys, pad, and stick lights up the backlight again as i hit them


i guess you can add in the readme that if one manually turns it off using fn+spacebar the service is interrupted and the service needs to be stopped then restarted again or simply hit fn+spacebar to manually turn it on and service will be in normal operation

perhaps this is just unique to me
please disregard


thank you

**Comment by VorpalBlade** [comment]
> if i do a manual cycle of fn+spacebar and never turn it back on, it interrupts the service(?) and wont behave normally

It is by design that it remembers and uses the level you set it to. E.g. if you set it to half-bright with Fn+Space, the service will switch between off and half-bright. Similarly if you set it to the brightest setting with Fn+Space, it will remember that and switch between off and full brightness.

If you switch it off with Fn+Space, that is an indication that the service you don't want it on currently. I usually turn it off completely if I'm outdoors during the day where it isn't needed.

It wouldn't be too difficult to add a setting to disable this: A new command line switch, not pass in a and some changes to the logic here: https://github.com/VorpalBlade/keyboard-backlightd/blob/a06dfcc386699d489cd2cc705485771f21befbc8/src/led.rs#L93-L99 
and here: https://github.com/VorpalBlade/keyboard-backlightd/blob/a06dfcc386699d489cd2cc705485771f21befbc8/src/main.rs#L59-L61

**Comment by VorpalBlade** [comment]
I believe the issue is now resolved, with improved documentation and a `--no-adaptive-brightness` flag.

The lack of RPM is not something I can fix (I have no idea how RPM packaging work). But I would welcome a PR that adds the ability to build a RPM.

I can help with the part to get it building automatically in CI for releases (if you are able to make it work on the Ubuntu LTS 22.04 build hosts that GitHub CI uses, possibly with docker containers if needed)
